src/nspawn/nspawn.c

```diff
@@ -1394,14 +1395,11 @@ static int setup_journal(const char *directory) {
         etc_machine_id = prefix_roota(directory, "/etc/machine-id");
 
         r = read_one_line_file(etc_machine_id, &b);
-        if (r == -ENOENT && try)
-                return 0;
-        else if (r < 0)
+        if (r < 0)
                 return log_error_errno(r, "Failed to read machine ID from %s: %m", etc_machine_id);
 
         id = strstrip(b);
-        if (isempty(id) && try)
-                return 0;
+        assert(!isempty(id));
```

Comment:
`assert` shouldn't be used for any external state. Even if the contents of the file were checked before, someone could externally modify it at any time.